cluster-operator/src/main/java/io/strimzi/operator/cluster/tracing/TracingUtils.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.tracing;
+
+import io.strimzi.api.kafka.model.tracing.JaegerOTelTracing;
+import io.strimzi.api.kafka.model.tracing.JaegerTracing;
+import io.strimzi.api.kafka.model.tracing.Tracing;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Handle any tracing abstraction
+ */
+public class TracingUtils {
+
+    private static final Map<Class<? extends Tracing>, TracingHandle> HANDLE_MAP;
+
+    static {
+        HANDLE_MAP = new HashMap<>();
+        HANDLE_MAP.put(JaegerTracing.class, new OpenTracingHandle());
+        HANDLE_MAP.put(JaegerOTelTracing.class, new OTelTracingHandle());
+    }
+
+    private static TracingHandle getTracingHandle(Tracing tracing) {
+        if (tracing == null) {
+            throw new IllegalArgumentException("null Tracing parameter");
```

Comment:
I would appreciate a better exception message

Reply:
Such as?
Afais, all we need is an indication that a null was passed in, and then you can deduct from stacktrace where it went wrong.

Reply:
looks much better now, thanks